### ISSUE TITLE
URL Tracker - Service Unavailable

### ISSUE THREAD

**Issue by StuartPaterson** [kipusoep/UrlTracker#187]
Hey,

Hope you can help us, on our Live Umbraco instance we're struggling to get access to URL Tracker,  seems to be timing out or something and we receive a 'Service Unavailable' error, although URL Tracker on our staging and local dev environments seem to be fine.

I've attached a file with the only entry I can find from the IIS logs around the time of the Service Unavailable Error and included are the Umbraco Trace logs around the same time.

Our environment is running - 

Umbraco version 7.5.11 assembly: 1.0.6274.27352 (upgrade being planned :) )
URL Tracker 3.11
IIS 8.5

[Logs.txt](https://github.com/kipusoep/UrlTracker/files/1942027/Logs.txt)

Thanks
Stuart

**Comment by kipusoep** [comment]
Try to clear up the icUrlTracker sql table. Remove entries where Is404=1 for example (logged notfound requests).

**Comment by StuartPaterson** [comment]
Cheers for this, Just checked though and we don't have any ls404 entries in the icUrlTracker sql table witha  value of 1

I'm looking through an export of the table just now but nothing obviously jumping out at me which could be causing issues.

**Comment by daniel-chenery** [comment]
Are you getting Service Unavailable in the backoffice, on the URL Tracker tab?

Your logs point to a problem with Image Processor, and I can't actually see a trace log for the URL Tracker (only the GET request)

How many entries are in your DB?
I presume it's related to #185/#184 - basically, we're pulling in too many records :(

**Comment by StuartPaterson** [comment]
Yeah it's on the URL Tracker tab Daniel, we have around 2,500 entries in each of our Umbraco instances, local, staging and live but only live has this issue.

**Comment by daniel-chenery** [comment]
Do staging and live have different SQL memory limits?
Or are they different SQL versions? E.G. Express/Web

If the databases and setup are identical, I can't (logically) think why it would cause a problem for one and not the other

**Comment by StuartPaterson** [comment]
SQL 2016 Std on Staging & Live with local running 2016 Dev Edition, no memory limitations and the same SQL config in place, I've just gone through them side by side.

I know, it has us completely stumped on this one...

**Comment by daniel-chenery** [comment]
This is great practice, but if the staging connects to live the DB does the same happen?

**Comment by StuartPaterson** [comment]
Good test Daniel, I'll see if we can can do this...taking backups beforehand of course :)

**Comment by StuartPaterson** [comment]
We've just managed to get a chance to do this test, good news is that connecting another Umbraco instance (same version/build) to the live database worked fine...still an issue on the live Umbraco instance though obviously.

One other thing I wanted to check, on our live server instance we seem to have around 5,500 files in the \App_Data\TEMP\ClientDependency folder for...do you think this could be causing a problem for URL Tracker? only 73 of the files relate to the current Client Dependency Handler version number.

Thanks
Stuart

**Comment by daniel-chenery** [comment]
The CD folder is built up of numerous things, such as Image Processor images, combined JS files and general Umbraco Cache
Now it **should** be safe to delete, but I can't see the Url Tracker being affected by this

**Comment by StuartPaterson** [comment]
Clearing out our cache is on our To Do list anyway so we'll see what happens
